File: algorithms/2_medium/N/02270_number_of_ways_to_split_array/prefix.c
int waysToSplitArray(const int *restrict nums, const int len) {
    if (len <= 1) { return 0; }

    int64_t suff = 0;
    for (int i = 0; i < len; i+= 1) { suff += nums[i]; }

    int64_t pref = 0;
    int cnt = 0;

     for (int i = 0; i < len - 1; i+= 1) {
        pref += nums[i];
        suff -= nums[i];

        if (pref >= suff) {
            cnt += 1;
        }
    }

    return cnt;
}

File: algorithms/2_medium/N/02270_number_of_ways_to_split_array/prefix.rs
impl Solution {
    pub fn ways_to_split_array(nums: Vec<i32>) -> i32 {
        if nums.len() <= 1 { return 0; }

        let len = nums.len();
        let mut suff: i64 = nums.iter().map(|x| *x as i64).sum();
        let mut pref: i64 = 0;
        let mut cnt: usize = 0;

        for &num in &nums[0..(len - 1)] {
            pref += num as i64;
            suff -= num as i64;

            if pref >= suff {
                cnt += 1;
            }
        }

        cnt.try_into().unwrap()
    }
}
